File: Java Script - Introduction to Programming/8.1. Exam Preparation - Part I/03. Point on Segment.js
function pointOnSegment([...arg]) {
  let [left, right, point] = [...arg.map(Number)]

  if (point >= Math.min(left, right) && point <= Math.max(left, right)) {
    console.log('in')
  } else {
    console.log('out')
  }

  if (Math.abs(left - point) < Math.abs(right - point)) {
    console.log(Math.abs(left - point))
  } else {
    console.log(Math.abs(right - point))
  }

}

// pointOnSegment(['10', '5', '7'])

File: Pyrhon - Introduction to Programming/8.1. Exam Preparation - Part I/03. Point on Segment.py
left, right, point = [int(input()) for _ in range(3)]

min_p = left if left < right else right
max_p = left if left > right else right
if min_p <= point <= max_p:
    print('in')
else:
    print('out')


if abs(left - point) < abs(right - point):
    print(abs(left - point))
else:
    print(abs(right - point))


'''
10
5
7
#in
#2
'''

